### PR TITLE
issue #2169 - change getValueConstant to getValueAsEnum and update project code

### PR DESCRIPTION
**Pull request by lmsurpre**
The first commit has the model change (getValueConstant  -> getValueAsEnum) and not much else.
I suggest focusing on the second commit for the review.

Signed-off-by: Lee Surprenant <lmsurpre@us.ibm.com>

### CODE REVIEW

File: fhir-model/src/main/java/com/ibm/fhir/model/util/FHIRUtil.java [hunk 1]
```diff
@@ -131,9 +131,9 @@ public static String toString(Visitable visitable) {
     private static Pattern buildReferencePattern() {
         StringBuilder sb = new StringBuilder();
         sb.append("((http|https)://([A-Za-z0-9\\\\\\/\\.\\:\\%\\$\\-])*)?(");
-        sb.append(Arrays.asList(ResourceType.ValueSet.values()).stream()
-            .map(v -> v.value())
-            .collect(Collectors.joining("|")));
+        sb.append(ModelSupport.getResourceTypes(false).stream()
+                .map(t -> ModelSupport.getTypeName(t))
+                .collect(Collectors.joining("|")));
```
Comment (prb112) [inline fhir-model/src/main/java/com/ibm/fhir/model/util/FHIRUtil.java:136]: Interesting... so we allowed Resource/1 and DomainResource/1 as valid references before. 
This is a good catch.

File: fhir-search/src/main/java/com/ibm/fhir/search/util/ReferenceUtil.java [hunk 2]
```diff
@@ -30,8 +30,8 @@
     private static final String URN = "urn:";
 
     // A set of resource type names to assist computing the base URL string
-    private static final Set<String> resourceTypes = Arrays.stream(FHIRResourceType.ValueSet.values())
-            .map(FHIRResourceType.ValueSet::value)
+    private static final Set<String> resourceTypes = Arrays.stream(FHIRResourceType.Value.values())
+            .map(FHIRResourceType.Value::value)
```
Comment (prb112) [inline fhir-search/src/main/java/com/ibm/fhir/search/util/ReferenceUtil.java:34]: I think there was an early piece of code that did a similar thing and used ModelSupport with the false.  Do we want that here too?

File: fhir-model/src/main/java/com/ibm/fhir/model/util/FHIRUtil.java (hunk 1 above)
Comment (lmsurpre) [reply fhir-model/src/main/java/com/ibm/fhir/model/util/FHIRUtil.java:136]: yes, very astute

**Review by prb112 — APPROVED** [review]
LGTM - left two comments... one about a nice change, and the other about applying the change in a different spot potentially.

File: fhir-search/src/main/java/com/ibm/fhir/search/util/ReferenceUtil.java (hunk 2 above)
Comment (lmsurpre) [reply fhir-search/src/main/java/com/ibm/fhir/search/util/ReferenceUtil.java:34]: sure, i can make that change

Comment (lmsurpre) [reply fhir-search/src/main/java/com/ibm/fhir/search/util/ReferenceUtil.java]: done

**Review by JohnTimm — APPROVED** [review]
LGTM

**Pull request merged**